libraries/redcore/api/hal/hal.php

```diff
@@ -628,6 +628,26 @@ public function apiCreate()
 		$functionName = RApiHalHelper::attributeToString($this->operationConfiguration, 'functionName', 'save');
 		$data = $this->triggerFunction('processPostData', $this->options->get('data', array()), $this->operationConfiguration);
 
+		// Remove primary keys for "model" and "table" data mode
+		$dataMode = RApiHalHelper::attributeToString($this->operationConfiguration, 'dataMode');
+
+		if (($dataMode == 'model') || ($dataMode == 'table'))
+		{
+			// Get the primary keys
+			$primaryKeys = $model->getTable()->getKeyName(true);
```

Comment:
We just prevent pass primary keys on Create method to avoid the update process. All posts data should send to method and I think get primary keys from table class is right way. We don't want client pass this parameter in POST method, that's why I don't want these primary keys doesn't exist in XML (client shouldn't know about this parameter).